src/homebridge-roku.js

```diff
@@ -100,29 +103,58 @@ class RokuAccessory {
     return accessoryInfo;
   }
 
+  doesSupportWakeOnLan() {
+    return this.info.supportsWakeOnWlan === 'true';
+  }
+
   setupTelevision() {
     const television = new Service.Television(this.name);
 
     television
       .getCharacteristic(Characteristic.Active)
-      .on('get', (callback) => {
-        this.roku
-          .info()
-          .then((info) => {
-            const value =
-              info.powerMode === 'PowerOn'
-                ? Characteristic.Active.ACTIVE
-                : Characteristic.Active.INACTIVE;
-            callback(null, value);
-          })
-          .catch(callback);
+      .on('get', async (callback) => {
+        try {
+          const info = await pTimeout(this.roku.info(), this.requestTimeout);
```

Comment:
> Looking at this again, will this ever work as expected?

It's been working for me so far 😛.

> If the TV is off, this will fail immediately and never trigger the TimeoutError check below.

Nope, `this.roku.info()` doesn't resolve until around 10 seconds later if the TV is offline; [node-fetch](https://www.npmjs.com/package/node-fetch) which is used by the Roku package sorta has [timeout support](https://www.npmjs.com/package/node-fetch#request-cancellation-with-abortsignal) but I don't think the Roku package implements it.